content/blog/2024-05-22-rustl-2024-unconference.md

```diff
@@ -0,0 +1,196 @@
+
+### Build tooling
+
+Everyone agreed that a better solution was needed to ship non-TUI Rust apps.
+
+There was discussion about various platforms and build system, and the difficulty of having to eg write Java code for an Android port, and whether anyone wanted to take responsibility for maintaining that glue layer for the rest of the ecosystem.
+
+Unfortunately the discussions on this subject didn't really go anywhere, as far as I could tell.
+Packaging Rust apps is still very much an open problem.
+
+### Text layout and editing
+
+There was some widespread interest in the Linebender ecosystem's text handling crates.
+From what I remember, Rik Arends and others had concerns about whether eg Parley and Swash depend on our Vello renderer (they don't).
+
+Some people also wanted Parley to be decoupled from Swash.
+
+People we especially interested in the idea of having access to a common text-editing widget, or at least infrastructure to create one.
+That widget would handle IME, text selection, accessibility actions, etc, in a way that would feel native to different platforms with their own text-editing quirks (mobile, MacOS, Linux with vim mode, etc).
+
+### Winit
+
+By now all of the Rust ecosystem has firmly converged on [`winit`](https://github.com/rust-windowing/winit) as the windowing platform-abstraction solution of choice.
+
+(Well, not all! [One small project](https://github.com/makepad/makepad) with indomitable maintainers still holds out against the invaders.)
+
+The consensus during this discussion seemed to be that Winit was there to stay, and that any changes we wanted would probably come in the form of upstream contributions, not competing projects.
+
+Glazier is still shelved and we're planning to port its features to Winit until it reaches feature parity.
+
+A discussion that came up was the idea of splitting more "vocabulary types" from Winit into standalone crates.
+Vocabulary crates, eg crates that only define types and very little code, have a very low cost in memory and build times, while allowing projects to interop with eg Winit without actually adding a dependency on Winit.
+
+This was connected with the discussions about text editing: having a vocabulary crate that would list Winit's events and also Parley's cursor movements would allow for a very low-dependency text editing widget.
+
+### Accessibility
+
+*Note: This section is also my personal opinion and doesn't represent the Linebender project either.*
+
+The subject of Accessibility came up informally, but I'll claim it was underexplored.
+
+While the subject was on the agenda, by the time we reached it the Unconference had been running long enough that everyone was tired and ready to leave or get to the catering.
+This is an unfortunate pattern that I think is worth pointing out: if projects don't make a deliberate effort to prioritize discussions of accessibility, the subject will be left out.
```

Comment:
For the sake of posterity, I'd like to point out that I did attempt to raise the topic publicly twice on Friday, since I wanted to share the results of our discussion with Matt in the office hours from Thursday evening; however, I was unable to do so (and also did not want to forcibly take over room focus) due to seemingly-productive ongoing private discussions that were already naturally occurring.

Note that I feel like this is sort of a damned-if-you-do, damned-if-you-don't situation. What you're asking for here seems to directly contradict the prior complaint about and the desire for a less-centralized discussion flow in favor of multiple smaller groups.

Reply:
> Note that I feel like this is sort of a damned-if-you-do, damned-if-you-don't situation.

That's not how I meant it, but it's a fair interpretation, and I need to make the text clearer.

To be clear, my position is that organizers are responsible for the IMO overly centralized format, but they're not responsible for the lack of discussion about accessibility. For the latter, I'd say the responsibility is collective, *precisely* because people expect it to be something brought to their attention and not something they have to think about spontaneously.